File: DP/도둑질/PGS_도둑질_이지영.java
public class PGS_도둑질_이지영 {

    class Solution {
        public int solution(int[] money) {
            int N = money.length;
            int[] first = new int[N+1];
            int[] last = new int[N+1];

            for (int i=2; i<=N; i++) {
                first[i] = Math.max(first[i-2]+money[i-2], first[i-1]);
                last[i] = Math.max(last[i-2]+money[i-1], last[i-1]);
            }

            return Math.max(first[N], last[N]);
        }
    }

}

File: DP/도둑질/PGS_도둑질_이지영.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> money) {
    int answer = 0;
    int N = money.size();
    int dp[1000001][2] = {0, };

    for (int i=2; i<=N; i++) {
        dp[i][0] = max(dp[i-2][0]+money[i-2], dp[i-1][0]);
        dp[i][1] = max(dp[i-2][1]+money[i-1], dp[i-1][1]);
    }

    return max(dp[N][0], dp[N][1]);
}